frontend/console/src/api/modules/use-modules.ts

```diff
@@ -38,7 +28,7 @@ export const useModules = () => {
   }
 
   return useQuery({
-    queryKey: [useModulesKey],
+    queryKey: [useModulesKey, schemaUpdatedAt],
```

Comment:
without this, the schema would change (delete old modules) and never trigger the modules to reload. This should be better once we have a dedicated streaming endpoint :)

Also note that with the current deployment/schema stuff, it's possible to have 2 active deployments for the same module for a few seconds while the old module gets cleaned up. We'll want to add handling to our streaming endpoint to return the most recently activated deployment vs. returning both the old and new.

In the future, this will need more attention since we'll likely want to show multiple deployed versions, but until we have versioning worked out we should probably just show the most recent active deployment.

Reply:
Nice, that makes sense. Thanks for catching that! The streaming endpoint does already filter for unique module names and chooses the latest when there is a dup. I haven't added update handling to the backend yet, though, so it's not yet tested.